pkg/machines/components/deleteDialog.jsx

```diff
@@ -103,16 +115,19 @@ export class DeleteDialog extends React.Component {
         Object.keys(vm.disks).sort()
                 .forEach(t => {
                     let d = vm.disks[t];
-                    if (d.type == 'file' && d.source.file)
-                        disks.push({ target: d.target, file: d.source.file, checked: !d.readonly });
+                    let entry = d;
+
+                    entry.checked = !d.readonly;
```

Comment:
This is just an alias for the same object, so the line below will actually modify the original `vm.disks` array. I suggest using the original `disks.push({ ... })` code with just the updated condition.

Reply:
Using the original ```disks.push({ target: d.target, file: d.source.file, checked: !d.readonly })``` will not work for other disks types apart from file ones.
Disks can be different types, they don't need to have always source.file attribute.